include/revng/Pypeline/Registrators/Detail/StringRefCaster.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+// This is a caster class, it allows nanobind to automatically convert python
+// strings to `llvm::StringRef`s and back
+
+#include "nanobind/nanobind.h"
+
+#include "llvm/ADT/StringRef.h"
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+NAMESPACE_BEGIN(detail)
+
+template<>
+struct type_caster<llvm::StringRef> {
+  NB_TYPE_CASTER(llvm::StringRef, const_name("str"))
+
+  bool from_python(handle Source, uint8_t, cleanup_list *) {
```

Comment:
I'm a bit concerned about the asymmetry of `from_python` and `from_cpp`.

The former returns a non-owning object.
The latter return an owning object.

Are you sure this the correct semantics fro a "caster"?

Reply:
I've copied the code almost 1:1 from [here](https://github.com/wjakob/nanobind/blob/master/include/nanobind/stl/string_view.h) so I'm pretty sure it's doing the right thing

Reply:
Done